src/Components/Samples/BlazorUnitedApp/Shared/MainLayout.razor

```diff
@@ -1,4 +1,5 @@
 ﻿@inherits LayoutComponentBase
+@implements IRazorComponentApplication<MainLayout>
```

Comment:
Ideally, we are going to source generate this interface definition and implementation based on a convention (like you have an App.razor file on your project) or another gesture (like extending a special type if we find a reason to have one)

Reply:
If that works out without drawbacks, that could be a good solution.

In many ways it's extremely strange that having a call to `DoSomething<T>` actually *causes* the type `T` to implement some other interface. We'll have to be super cautious about only considering this in cases where we can be completely sure it will never cause some other compiler error because people would be totally mystified about what's going on. And I'm not sure how it will make sense if `T` is not in the current project.